source/elements/oneDAL/source/glossary.rst

```diff
@@ -13,92 +13,113 @@ Machine learning terms
-        A process of applying a :term:`trained<Training>` :term:`model` to the :term:`inference set` with a goal to
-        make outputs based on input :term:`feature vectors<Feature vector>`.
+        A process of applying a :term:`trained<Training>` :term:`model`
+        to the :term:`dataset` with a goal to make predictions of
+        :term:`responce` values based on input :term:`feature vectors<Feature vector>`.
 
         Synonyms: prediction.
 
     Inference set
-        An unlabeled :term:`dataset` used at the :term:`inference` stage to make an outputs from the :term:`model`.
+        A :term:`dataset` (usually without :term:`responses<Response>` present)
+        used at the :term:`inference` stage.
 
     Label
-        A :term:`feature` of some object or event which value is target to be predicted in :term:`supervised learning` task.
-        On the :term:`training` stage, the model which finds the best dependency between input :term:`feature vector` and
-        :term:`labels<Label>` is selected. :term:`Label` is one of outputs can be made by :term:`model`
-        on the :term:`inference` stage.
+        A :term:`response` with :term:`categorical<Categorical feature>` or :term:`ordinal<Ordinal feature>`
+        values. This is an output in :term:`classification` and :term:`clustering` tasks.
+
+        Examples: the spam-detection task has a binary label indicating whether the email is spam or not.
 
     Model
         A result of :term:`training` machine-learning algorithm that stores information necessary to run
         :term:`inference` on a new :term:`dataset`.
 
+        Examples: in linear regression algorithm, the model contains weight values for each input feature
+        and single bias value.
+
     Observation
-        A vector of at least one :term:`feature` value and zero or more :term:`label<Label>` values.
-        Each of :term:`labels<Label>` and :term:`features<Feature>` can encode different objects or events.
+        A vector of at least one :term:`feature` and zero or more :term:`responses<Response>`.
+        Each of them can encode different objects or events.
 
         Synonyms: instance, sample.
 
     Ordinal feature
         A :term:`feature` with a discrete set of possible values and some ordering between them.
 
+        Examples: student's grade.
+
     Regression
         A :term:`supervised machine-learning problem<Supervised learning>` of assigning
-        :term:`continuous<Continuous feature>` :term:`labels<Label>` to
+        :term:`continuous<Continuous feature>` :term:`responses<Response>` to
         :term:`feature vectors<Feature vector>`.
 
+        Examples: prediction of the temperature based on weather conditions.
+
     Response
-        A value of a single :term:`label`. For example, response may be a flag
-        indicating that an observation belongs to a particular class.
+        A property of some real object or event which value is target to be predicted in the
```

Comment:
is target to be predicted - we need a help of Ekaterina to word it in a cleaner way

Reply:
@outoftardis , please, help :)

Reply:
sure, I'm reviewing the changes now and will add there a suggestion how to rephrase it